apps/site/lib/site_web/controllers/schedule/line_api.ex

```diff
@@ -53,18 +53,13 @@ defmodule SiteWeb.ScheduleController.LineApi do
     json(conn, nil)
   end
 
-  @doc """
-  Provides predictions and vehicle information for a given route and direction, organized by stop.
-  The line diagram polls this endpoint for its real-time data.
-  """
-  @spec realtime(Plug.Conn.t(), map()) :: Plug.Conn.t()
-  def realtime(conn, %{"id" => route_id, "direction_id" => direction_id}) do
+  defp do_realtime(route_id, direction_id, date, now) do
```

Comment:
I think it'd make the data flow easier to follow if we placed "implementation detail" functions below the higher-level functions that call them — do you think that's a good convention?

Reply:
Yes, I like to adhere to that when possible, will move the function